Lumina node implementation

gabros20 opened the pull request.

dabit3 commented on src/pages/markdown-pages/learn/modular-architectures.md:
```diff
@@ -17,65 +17,56 @@ description: "A look into the different ways in which modular blockchains can be
 #### Introduction
 
-A new paradigm is emerging in which modular blockchains are enabling new chains to be constructed in ways that were not previously possible. Because of this, the design possibilities are vast for both the individual blockchain and the modular stack it is a part of. Different types of modular blockchains can work synergistically, varying by purpose and architecture.
+A new paradigm is emerging in which modular blockchains are enabling new chains to be constructed in ways that were not previously possible. Because of this, the design possibilities are vast for both the individual blockchain and the modular stack it is a part of. Different types of modular blockchains can work collaborative manner, varying by purpose and architecture.
```
maybe "can work in a collaborative manner"

dabit3 commented on src/pages/markdown-pages/learn/modular-scalability.md:
```diff
@@ -17,37 +17,39 @@ description: "The scalability properties of different types of modular blockchai
+#### Summary
+1. The two main designs of rollups are optimistic and zk-Rollups. Optimistic rollups provide scalability through data compression and alleviating state growth on the base layer. zk-Rollups scale similarly, as well as through efficient compute verification with validity proofs.
+2. Data availability layers scale block verification and minimize bandwidth costs for light nodes. This is possible with data availability sampling, which allows light nodes to verify data availability by only downloading small random samples of a block.
+3. Settlement layers can scale using the same techniques from execution and data availability layers. In their current form, settlement layers are unideal. A more optimal approach would be to create an environment that limits smart contracts to favor interactions with execution layers.
```
Is this essentially saying a settlement layer optimized for rollups / batch transactions? if so there might be some opportunity to be more clear about this.

dabit3 commented on src/pages/markdown-pages/learn/modular-scalability.md:
```diff
@@ -68,15 +70,4 @@ The current scope of settlement layers has been borne out of necessity for a blo
-2. Data availability layers scale block verification and minimize bandwidth costs for light nodes. This is possible with data availability sampling, which allows light nodes to verify data availability by only downloading small random samples of a block.
-3. Settlement layers can scale using the same techniques from execution and data availability layers. In their current form, settlement layers are unideal. A more optimal approach would be to create an environment that limits smart contracts to favor interactions with execution layers.
-</div>
+The vast majority of the settlement layers transaction activity should consist of execution layers settling transactions. <a href="https://forum.celestia.org/t/increasing-scalability-of-the-evm-for-rollups-by-restricting-state-and-contract-set/78" target="_blank" rel="noopener noreferrer" style="color:#7B2BF9;">One method</a> of influencing this behavior is to limit smart contracts such that they ...
```
This is such a great paragraph.

dabit3 reviewed (approved):
Looks great, left a few minor comments.

alex-beckett commented:
thx for the review ser @dabit3

gabros20 committed:
feat: Add NodeStatus component and related SVG icons

- Created NodeStatus component for displaying block verification status
- Added LuminaCheckmarkSVG, LuminaDiagonalArrowSVG, LuminaErrorSVG, and LuminaGradientCircleSVG
- Integrated NodeStatus into DesktopNav
- Updated .gitignore to exclude docs/ directory

gabros20 committed:
chore: Update Next.js config and SVG components

- Add Next.js configuration with page extensions, image optimization, and redirects
- Fix SVG components by updating SVG attribute syntax (stroke-* to strokeLinecap/strokeLinejoin)
- Migrate inline styles to React object syntax in LuminaGradientCircleSVG
- Sanitize JSON-like attributes in SVG components

gabros20 committed:
feat: working connection

gabros20 committed:
fix: loader svg animation direction

gabros20 committed:
fix: updated loader svg

gabros20 committed:
fix: multiple hydration issues in the original project build

gabros20 committed:
feat: Enhance NodeStatus component with dynamic syncing progress and improved status messaging

gabros20 committed:
chore: Refactor blog fetching, NodeStatus, and navigation components

- Improve blog post fetching with error handling and caching
- Update NodeStatus component styling and progress display
- Relocate AutoConnectingLuminaNode component
- Simplify LuminaDiagonalArrowSVG icon

gabros20 committed:
feat: Enhance animation and navigation components

- Refactor MovingGradient component to generate random gradient animations on client-side
- Update AutoConnectingLuminaNode to manage initialization state
- Modify NodeStatus to include animation controls and improved syncing messages
- Introduce MenuButton component for mobile navigation with SVG icons
- Clean up Nav component to integrate MenuButton and streamline layout

gabros20 committed:
fix: wasm issues build time

gabros20 committed:
feat: Update NodeStatus and useLuminaNode for improved syncing display

- Refactor NodeStatus to utilize syncProgress for approximate syncing percentage display as much possible
- Remove fake progress logic and simplify state management
- Enhance useLuminaNode hook to calculate and provide sync progress and connected peers
- Update event handling for better status updates during syncing and connection

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
Update package-lock.json and EcosystemExplorer.js: bump package versions and clean up imports

gabros20 committed:
fix: connection issues

gabros20 committed:
fix: inifinie loop

gabros20 committed:
fix loading state indication

gabros20 committed:
feat: working block loading

gabros20 committed:
feat: new block rendering component added

gabros20 committed:
Merge branch 'feature/lumina-node-v2' into feature/lumina-node-integration

gabros20 committed:
commdent

gabros20 committed:
fix: live block render

gabros20 committed:
fix: build error

gabros20 committed:
Merge branch 'feature/test-lumina-node' into feature/lumina-node-integration

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
fix: animations
